openslides_backend/action/actions/topic/json_upload.py

```diff
@@ -0,0 +1,136 @@
+from ....permissions.permissions import Permissions
+from ....shared.interfaces.event import Event, EventType
+from ....shared.interfaces.write_request import WriteRequest
+from ....shared.patterns import fqid_from_collection_and_id
+from ....shared.schema import required_id_schema
+from ...action import Action
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionResultElement
+from ..agenda_item.agenda_creation import agenda_creation_properties
+from .create import TopicCreate
+from .mixins import DuplicateCheckMixin
+
+
+@register_action("topic.json_upload")
+class TopicJsonUpload(DuplicateCheckMixin, Action):
+    """
+    Action to allow to upload a json. It is used as first step of an import.
+    """
+
+    model = Topic()
+    schema = DefaultSchema(Topic()).get_default_schema(
+        additional_required_fields={
+            "data": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        **model.get_properties("title", "text"),
+                        **{
+                            prop: agenda_creation_properties[prop]
+                            for prop in (
+                                "agenda_comment",
+                                "agenda_type",
+                                "agenda_duration",
+                            )
+                        },
+                    },
+                    "required": ["title"],
+                    "additionalProperties": False,
+                },
+                "minItems": 1,
+                "uniqueItems": False,
+            },
+            "meeting_id": required_id_schema,
+        }
+    )
+    permission = Permissions.AgendaItem.CAN_MANAGE
+    headers = [
+        {"property": "title", "type": "string"},
+        {"property": "text", "type": "string"},
+        {"property": "agenda_comment", "type": "string"},
+        {"property": "agenda_type", "type": "string"},
+        {"proptery": "agenda_duration", "type": "number"},
+    ]
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        data = instance.pop("data")
+
+        # enrich data with meeting_id
+        for entry in data:
+            entry["meeting_id"] = instance["meeting_id"]
+
+        # validate and check for duplicates
+        self.init_duplicate_set(instance["meeting_id"])
+        self.rows = [self.validate_entry(entry) for entry in data]
+
+        # generate statistics
+        itemCount, itemNew, itemError = 0, 0, 0
+        for entry in self.rows:
+            itemCount += 1
+            if entry["status"] == "new":
+                itemNew += 1
+            if entry["status"] == "error":
+                itemError += 1
+        self.statistics = {
+            "itemCount": itemCount,
+            "Created": itemNew,
+            "Error": itemError,
```

Comment:
Do I understand it correctly that these strings are to be displayed as-is in the client? If so, we should choose more suitable ones.

Reply:
Use Total/New/Error now. Or do you have a better idea?

Reply:
Use Luisa's, see https://github.com/OpenSlides/openslides-backend/issues/1661#issuecomment-1466109237, like
* total: 10
* created: 5
* omitted: 5